src/PsrLoggerAbstractAdapterFactory.php

```diff
@@ -5,25 +5,80 @@
 namespace Laminas\Log;
 
 use Interop\Container\ContainerInterface;
+use Interop\Container\Exception\ContainerException;
+use Laminas\ServiceManager\AbstractFactoryInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
 
 /**
  * PSR Logger abstract service factory.
  *
  * Allow to configure multiple loggers for application.
  */
-class PsrLoggerAbstractAdapterFactory extends LoggerAbstractServiceFactory
+class PsrLoggerAbstractAdapterFactory implements AbstractFactoryInterface
 {
     /**
      * Configuration key holding logger configuration
-     *
-     * @var string
      */
-    protected $configKey = 'psr_log';
+    protected string $configKey = 'psr_log';
+
+    private LoggerAbstractServiceFactory $loggerAbstractServiceFactory;
+
+    public function __construct()
+    {
+        $this->loggerAbstractServiceFactory = new LoggerAbstractServiceFactory($this->configKey);
+    }
 
-    public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null)
+    /**
+     * @param string $requestedName
+     * @throws ContainerException
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null): PsrLoggerAdapter
     {
-        $logger = parent::__invoke($container, $requestedName, $options);
+        $loggerFactory = $this->loggerAbstractServiceFactory;
+        $logger        = $loggerFactory($container, $requestedName);
 
         return new PsrLoggerAdapter($logger);
     }
+
+    /**
+     * Determine if we can create a service with name
+     *
+     * @param string $name
+     * @param string $requestedName
+     */
+    public function canCreateServiceWithName(ServiceLocatorInterface $serviceLocator, $name, $requestedName): bool
+    {
+        return $this->loggerAbstractServiceFactory->canCreate($serviceLocator, $name);
+    }
```

Comment:
@Ocramius 
laminas-log uses laminas-servicemanager 3 and methods like these are from version 2 of laminas-servicemanager. In other components this has always been removed, no matter if the factory class was final or not. Should the same approach be taken here?

Reply:
We should certainly push for v3 API.

I didn't check this yet, only skimmed and saw no tests, hence my initial feedback :)

Reply:
Sure, I will remove v2 methods, very happy about that :+1: